logstash-core/lib/logstash/agent.rb

```diff
@@ -10,80 +10,125 @@
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
-  attr_reader :logger
+  attr_reader :logger, :state
 
-  def initialize(logger)
-    @logger = logger
+  def initialize(params)
+    @logger = params[:logger]
     @pipelines = {}
+    @pipeline_threads = {}
+    @state = clean_state
```

Comment:
This is confusing because its not a `state` like `:running`, but a config string. I'd like a name that better describes that. Maybe `current_config`?